lib/Driver/DarwinToolChains.cpp

```diff
@@ -718,15 +746,33 @@ static void validateDeploymentTarget(const toolchains::Darwin &TC,
   }
 }
 
+static void validateTargetVariant(const toolchains::Darwin &TC,
+                                  DiagnosticEngine &diags,
+                                  const llvm::opt::ArgList &args,
+                                  StringRef defaultTarget) {
+  if (TC.getTargetVariant().hasValue()) {
+    auto target = TC.getTriple();
+    auto variant = *TC.getTargetVariant();
+
+    if (!triplesAreValidForZippering(target, variant)) {
+      diags.diagnose(SourceLoc(), diag::error_unsupported_target_variant,
+                    variant.str(),
+                    variant.isiOS());
```

Comment:
I'm working on porting this to swift-driver, and I've found that I don't understand this diagnostic. In particular, I think that if you pass `-target x86_64-apple-ios13 -target-variant x86_64-apple-macos10.15`, it will end up emitting an error claiming that `-target` is `x86_64-apple-macos10.15`. (The tests wouldn't catch this because they match this part of the diagnostic with a `{{.*}}`.) Am I misinterpreting the diagnostic, or should we be passing the value of `target.str()` rather than `variant.str()` when `!variant.isiOS()`?